search-api/src/main/java/edu/unc/lib/boxc/search/api/requests/SearchState.java

```diff
@@ -323,6 +327,16 @@ public String toString() {
             }
             return leftHand + "," + rightHand;
         }
+
+        private void checkValidRangePair(String start, String end) {
+            // As long as one side is set we're good to go.
+            if (start != null && end != null) {
+                if (Integer.parseInt(start) >= Integer.parseInt(end)) {
```

Comment:
if either start or end is null then `Integer.parseInt` will throw a NumberFormatException. So I think we'll need to individually check each variable before parsing it. We may need to add some unit tests for RangePair so we can make sure its behaving as expected (breaking it out into its own file would be fine as well). 

We need to verify:
1. if either start or end is set to anything other than null or a number, that it will throw an exception (IllegalArgumentException is fine)
2. leftHand can be set to a valid value while rightHand is set to null, via the setters AND the string splitting constructor (plus the reverse)
3. if leftHand is greater than rightHander, that an IllegalArgumentException is thrown. Need to test this via setters and constructor.
4. that both the left and right hand can be set to valid integers

Reply:
In this case shouldn't it avoid trying to parse null, since it only runs the check if both sides have values?